draft-ietf-ppm-dap.md

```diff
@@ -484,6 +480,132 @@ struct {
 } HpkeCiphertext;
 ~~~
 
+## Queries {#query}
+
+DAP is designed to give the Collector control over which measurements are
+included in a batch for aggregation. However, since reports are uploaded to the
+Leader, the Collector cannot choose the batch directly. Instead, it issues a "query" used
+by the Aggregators to guide batch selection.
+
+This document defines the following query types:
+
+~~~
+enum {
+   reserved(0), // Reserved for testing purposes
+   time-interval(1),
+   fixed-size(2),
+   (65535)
+} QueryType;
+~~~
+
+Additional query types can be defined as needed. A query includes parameters
+used by the Aggregators to select a given batch of measurements specific to the
+given query type:
+
+~~~
+enum {
+    next-batch(0),
+    by-batch-id(1),
+} FixedSizeQueryType;
+
+opaque BatchId[32];
+
+struct {
+    FixedSizeQueryType fixed_size_query_type;
+    select (fixed_size_query_type) {
+        next-batch:  Empty;
+        by-batch-id: BatchId batch_id;
+    }
+} FixedSizeQuery;
+
+struct {
+    select (query_type) { // determined by query configuration
+        case time-interval: Interval batch_interval;
+        case fixed-size: FixedSizeQuery fixed_size_query;
+    }
+} Query;
+~~~
+
+The query is issued in-band as part of the collect sub-protocol
+({{collect-flow}}). Its content is determined by the "query type", which in turn
+is encoded by the "query configuration" configured out-of-band:
+
+~~~
+struct {
+    QueryType query_type;
+    Duration time_precision;
+    uint64 min_batch_size;
+    select (query_type) {
+        case time-interval:
+            Empty;
+        case fixed-size:
+            uint64 max_batch_size;
+    }
+} QueryConfig;
```

Comment:
We need to define what all these fields mean. For example, what is the `time_precision` field?

Reply:
I've added a line to punt to the relevant subsection. Will this suffice?
> The parameters pertaining to each query configuration are described in the relevant the subsection below.

Reply:
That's fine, yeah.